### DISCUSSION TITLE
exported image color

### DISCUSSION THREAD

**Discussion by lkevinzc** [vinothpandian/react-sketch-canvas#33] [Q&A] [answered]
Hi! In the `ReactSketchCanvas` we can set the `strokeColor`, but when I call the `.current.exportImage()` to get the base64 image and display using python, seems that its color distribution contains some other values.

For example, I only use `#ff0000` and `#00ff00` for my stroke color, and the exported image is like the left, but when I look at the r channel distribution, it's like the right:
| exported image | color distribution|
| --- | --- |
![image](https://user-images.githubusercontent.com/38581401/129843482-4c5d7d94-f834-4825-bc88-2ec8aa16b1ec.png) | ![image](https://user-images.githubusercontent.com/38581401/129846444-67ee5c66-e2b0-4118-be63-64a14b20538a.png)

May I know why this could happen?

**Comment by vinothpandian** [comment, marked as answer]
That's a very interesting find. The underlying logic to convert SVG sketch to PNG for download is by rendering it on the web canvas. This issue might be due to rasterization. 

Can you try exporting it directly as SVG and then rasterizing it in python? Maybe you can use svg2png function in cairosvg for better rasterization.

**Reply by lkevinzc** [reply]
Thanks for the explanation. The rasterization inaccuracy makes sense and I guess only the pixel values at boundary are affected. This should be fine after some thresholding to get binary values. 

Thanks again!
